Source/Core/Core/HW/GCMemcard/GCMemcard.cpp

```diff
@@ -1287,7 +1376,7 @@ s32 GCMemcard::FZEROGX_MakeSaveGameValid(const Header& cardheader, const DEntry&
     return 0;
 
   // get encrypted destination memory card serial numbers
-  cardheader.CARD_GetSerialNo(&serial1, &serial2);
+  auto[serial1, serial2] = cardheader.CalculateSerial();
```

Comment:
Every open source implementation I've seen get's this very wrong, the serial number isn't two 32 bit values, it's a 64bit value. I know this because of how MP1 utilizes 64bit values for it's collision materials and the behavior is identical to this:
https://gitlab.axiodl.com/AxioDL/kabufuda/blob/master/lib/kabufuda/Card.cpp#L725-733

Reply:
I can't see how it matters, it's likely just the result of 64bit variable being lowered to gc ppc.